dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/RelationshipRestRepository.java

```diff
@@ -60,4 +91,85 @@ public RelationshipRest findOne(Context context, Integer integer) {
     public DSpaceResource<RelationshipRest> wrapResource(RelationshipRest model, String... rels) {
         return new RelationshipResource(model, utils, rels);
     }
+
+    protected RelationshipRest createAndReturn(Context context, List<DSpaceObject> list)
+        throws AuthorizeException, SQLException, RepositoryMethodNotImplementedException {
+
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+
+        Relationship relationship = new Relationship();
+        if (list.size() == 2 && list.get(0).getType() == Constants.ITEM && list.get(1).getType() == Constants.ITEM) {
+            Item leftItem = (Item) list.get(0);
+            Item rightItem = (Item) list.get(1);
+            RelationshipType relationshipType = relationshipTypeService
+                .find(context, Integer.parseInt(req.getParameter("relationshipType")));
+
+            EPerson ePerson = context.getCurrentUser();
+            if (authorizeService.authorizeActionBoolean(context, leftItem, Constants.WRITE) ||
+                authorizeService.authorizeActionBoolean(context, rightItem, Constants.WRITE)) {
+                relationship.setLeftItem(leftItem);
+                relationship.setRightItem(rightItem);
+                relationship.setRelationshipType(relationshipType);
+                relationship = relationshipService.create(context, relationship);
+                return relationshipConverter.fromModel(relationship);
+            } else {
+                throw new AccessDeniedException("You do not have write rights on this relationship's items");
+            }
+        } else {
+            throw new UnprocessableEntityException("The given items in the request were not valid items");
+        }
+
+
+    }
+
+    @Override
+    protected RelationshipRest put(Context context, HttpServletRequest request, String apiCategory, String model,
+                                   Integer id, List<DSpaceObject> dSpaceObjects)
+        throws RepositoryMethodNotImplementedException, SQLException, AuthorizeException {
+
+        Relationship relationship = relationshipService.find(context, id);
+        if (relationship == null) {
+            throw new ResourceNotFoundException(apiCategory + "." + model + " with id: " + id + " not found");
+        }
+
+        if (dSpaceObjects.size() == 2 && dSpaceObjects.get(0).getType() == Constants.ITEM
+            && dSpaceObjects.get(1).getType() == Constants.ITEM) {
+            Item leftItem = (Item) dSpaceObjects.get(0);
+            Item rightItem = (Item) dSpaceObjects.get(1);
+
+            if (authorizeService.authorizeActionBoolean(context, leftItem, Constants.WRITE) ||
+                authorizeService.authorizeActionBoolean(context, rightItem, Constants.WRITE)) {
```

Comment:
Unlike the other comments in this class, in this situation, the `relationshipService.update()` currently seems to be *requiring* you to be an Admin.  So, I think we need to update `relationshipService.update()` to check for WRITE permissions, and then we might be able to remove the checks from here.

Reply:
@tdonohue we would need to discuss the permissions required to update the items in a relationship. I would consider this to be an admin-only feature.
Please ignore that Raf updated it, we will revert the update once we discussed this in detail in the next meeting

Reply:
We discussed the permissions yesterday, and it has been decided that the user should have BOTH:
* WRITE permissions on one of the current 2 items
* WRITE permissions on one of the new 2 items
It has been adjusted accordingly, and ITs has been created to verify the permissions